apollo-portal/src/main/resources/static/scripts/controller/IndexController.js

```diff
@@ -251,10 +135,25 @@ function IndexController($scope, $window, $translate, toastr, AppUtil, AppServic
         NamespaceService.find_public_namespaces()
             .then(function (result) {
                 $scope.allPublicNamespaces = result;
```

Comment:
It looks like the current implementation is to load all the data and do the pagination and filter in javascript. It is ok for most of the cases, but if there are a large amount of data, then it might make the browser very slow. What do you think?

Reply:
I found the new namespace page also loaded all the public namespaces data and its performance is ok in our [demo site](http://106.54.227.205/namespace.html?#/appid=100004458) so I think we could keep the current implementation.